docs/proposals/secret-annotation-new-app.md

```diff
@@ -0,0 +1,60 @@
+Design proposal to implement [Use build source secret based on annotation on
+Secret](https://trello.com/c/NoVpS2OS/1059-5-use-build-source-secret-based-on-annotation-on-secret-builds)
+
+(The card goal is for oc new-app/new-build to discover a relevant pre-existing
+Secret and attempt to use it when a source repo requires authentication.  The
+intention is to do this without adding yet more CLI arguments to
+new-app/new-build.)
+
+
+Design proposal:
+----------------
+
+- A Secret may include an annotation with a well-known name.  The name
+  *io.openshift.match-urls* is proposed.
```

Comment:
suggest including "clone", "git", or "source" in the name since this secret will be used for git source cloning and we may later want to introduce a distinct way to annotate secrets for docker operations and they might overlap the same URIs (at least same domain).